posts/2023-11-17-tictactoe-tutorial.mdx

```diff
@@ -0,0 +1,665 @@
+---
+layout: post
+name: 'Building a TicTacToe with Waku'
+title: 'Building a TicTacToe with Waku'
+date: 2023-11-29 00:00:00
+authors: [guru]
+published: true
+slug: tictactoe-tutorial
+categories: waku, tutorials
+image: /img/black-waku-logo-with-name.png
+hide_table_of_contents: false
+---
+
+# Building a Tic-Tac-Toe game with Waku
+
+In this beginner-friendly tutorial, we will be building a basic TicTacToe game that can be played by two players who are connected using Waku.
+
+<!--truncate-->
+
+![header](https://ucarecdn.com/044637ee-696f-4328-933b-3ccbc5581cbd/Screenshot20231129at22623AM.png)
+
+## What is Waku
+
+Waku is a family of robust, censorship-resistant, and privacy-preserving communication protocols designed to enable perfomant messaging for web3 apps. When I say ‘messaging’, this does not restrict Waku’s use cases just to building ‘text’ messaging apps; Waku can also create other use cases like games, marketplaces, governance tooling, and much more.
+
+If you wish to know more about what Waku offers - check out the [website](https://waku.org) or [developer documentation](https://docs.waku.org).
+
+## Why Waku for building a Tic-Tac-Toe game?
+
+A Tic-Tac-Toe game is one of the most straightforward project ideas that can be implemented with Waku. This is because it involves just two players and has minimal web3 jargon. Implementing the game is very clear for anyone with prior experience building with React-based frameworks and some enthusiasm to learn Waku.
+
+## Getting started with building
+
+### Prerequisites
+
+- Some intermediate-level knowledge of React based frameworks (In this tutorial, we use NextJS)
+- Understanding of Waku's architecture
+- Some coffee to sip along
+
+### Wireframing
+
+Before we start, let us visualise the project's wireframe for a better understanding: 
+
+![Wireframe](https://ucarecdn.com/5dcf3cca-cfa3-4d26-9a87-960e08e35a9e/wireframe.jpeg)
+
+The first screen on the left will be our landing screen with two buttons for starting a new game or joining an existing one. Each of the games in our TicTacToe game will have a `room-id` which two players will be connected to.
+
+Waku follows a 'content topic based architecture', meaning that the messages are routed and stored in different content topics across the network. In our case, the content topics would be the `room-id` since they serve the same purpose. Two players in a game will be connected on the same room-id.
```

Comment:
I would maybe drop in some technical terms here with regards to "being connected"

```suggestion
Waku follows a 'content topic based architecture', meaning that the messages are routed and stored based on different content topics across the network. In our case, the content topic would be the `room-id` since it maps perfectly. Two players in a game will be connected by publishing on and subscribing to the same content-topic/`room-id`.
```